src/components/CircularOutline/CircularOutline.js

```diff
@@ -1,132 +1,53 @@
 // @flow
 /**
- * Draws a totally-rounded rectangle inside the parent node.
- *
- * NOTE: This turned out to be a surprisingly hard problem! I had to use an
- * SVG since I needed a gradient border, and because the SVG is computed
- * dynamically, a few re-renders need to take place.
- *
- * Here's the flow:
- *   - Initial render:  Capture a reference to the SVG, so we can work out the
- *                      width and height of the parent container
- *
- *   - Path length:     Now that we have the length/height, we can add our SVG
- *                      <rect> that fills the space. After that update is
- *                      flushed to the DOM, we can figure out the length of the
- *                      path with `getTotalLength`.
- *                      TODO: I can probably use math to work out that path
- *                      so that I can skip this update cycle?
- *
- *   - Enabling trans:  We don't want the 'self-drawing' transition to happen
- *                      on mount. Because of this, transition starts as
- *                      disabled.
- *                      Once we make it to this stage, though, we can set
- *                      `finishedAllMountingSteps` to indicate that all this
- *                      stuff is done, so that any future changes to `isShown`
- *                      can be animated
- *
- * There's surely a lot of room for improvement with this flow.
+ * Draws a totally-rounded circle inside the parent node.
  */
 import React, { Component } from 'react';
 import styled from 'styled-components';
-import { Motion, spring } from 'react-motion';
+import { Spring, animated } from 'react-spring';
 
 import { COLORS } from '../../constants';
 
 type Props = {
+  size: number,
   color1: string,
```

Comment:
I know this is a bit unrelated but how do you feel about changing `color1` and `color2` to `colors` so that it matches the API of the other Buttons?

Reply:
The only issue is that because it's an SVG gradient, it's not as trivial to set up (not that it's super hard, but you'd need to add stops for each color provided, and work out the percentages for each stop)

I could use Flow to enforce that it's a tuple of 2 colors, so that the API matches but it's a limited number of entries... but I'm not sure how many folks run Flow in realtime, so it might not help them if they're confused why the third color isn't showing up. Although I don't expect us to ever need 3 colors for circular outlines anyway, so maybe that's not a real concern?

Reply:
Don't this it would be too hard to support an arbitrary number of stops: #266

Reply:
Right, so I didn't mean that it would be genuinely challenging, just that it's maybe not worth the time given that we have no need of 3+ color gradients.

Your PR looks like it'd work to me, but it also adds quite a bit of complexity to the code - it's a bit harder to follow with a dynamic number of stops. I'm not sure how it handles edge cases either - for example, if the colors I supply are `['#FFF']`, I won't get a solid white edge, but a gradient from white to purple.

I think if we wanted to go this route, we should add an `SvgGradient` component which can handle any number of colors, add some stories for it (well, timing's a bit awkward as we move to docz) to make sure it handles edge-cases.

I think we should create a new issue for it, with the "Good first issue" tag. Maybe someone would be interested in picking it up?

Reply:
Pretty sure it’s handling `[‘#FFF’]` as it should (taking the latest color stop for 100% which would be white as well) but I see what you mean

Reply:
> Pretty sure it’s handling [‘#FFF’] as it should (taking the latest color stop for 100% which would be white as well) but I see what you mean

Ah, you're right! Sorry, probably should have actually tried the code before commenting. Tried to get a reply out quickly between tasks at work.

I do like the idea of a consistent API, and it's probably worth the few minutes testing your PR + updating existing callsites. I'll try and get this done tomorrow morning, although I'm not sure how much time I'll have. I'll also create a Good First Issue ticket to extract a gradient component with proper stories and such